Community/Tdarr_Plugin_advo_FFMPEG_NVENC_Tiered_MKV_with_rollback.js

```diff
@@ -0,0 +1,563 @@
+      response.processFile = false;
+      response.infoLog += '☑File is already in hevc! \n';
+      return response;
+    }
+
+    response.preset = '-c:v hevc_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'h264') {
+    if (bitrateprobe == null || bitrateprobe < 100000) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+    if ((file.video_resolution === '480p' || file.video_resolution === '576p')
+      && (bitrateprobe / 1024) < inputs.h264BitrateCutoff * 0.5) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+    if (file.video_resolution === '720p' && (bitrateprobe / 1024) < inputs.h264BitrateCutoff * 0.75) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+
+    if (file.video_resolution === '1080p' && (bitrateprobe / 1024) < inputs.h264BitrateCutoff) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+    if (file.video_resolution === '4KUHD' && (bitrateprobe / 1024) < inputs.h264BitrateCutoff * 1.5) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+
+    if (parseInt(videoTrack.BitDepth, 10) > 8) {
+      // Remove HW Decoding for 10 bit
+    } else {
+      response.preset = '-c:v h264_cuvid';
+    }
+  } else if (file.video_codec_name.toLowerCase() === 'mjpeg') {
+    response.preset = 'c:v mjpeg_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'mpeg1') {
+    response.preset = '-c:v mpeg1_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'mpeg2') {
+    response.preset = '-c:v mpeg2_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'mpeg4') {
+    response.preset = '-c:v mpeg4_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'vc1') {
+    response.preset = '-c:v vc1_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'vp8') {
+    response.preset = '-c:v vp8_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'vp9') {
+    response.preset = '-c:v vp9_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'av1') {
+    response.preset = '-c:v av1_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'vc1') {
+    response.preset = '-c:v vc1_cuvid';
+  }
+
+  for (let i = 0; i < file.ffProbeData.streams.length; i += 1) {
+    // strip streams not supported by mkv
+    try {
+      if (
+        file.ffProbeData.streams[i].codec_name.toLowerCase() === 'eia_608'
+        || file.ffProbeData.streams[i].codec_name.toLowerCase() === 'timed_id3'
+        // || file.ffProbeData.streams[i].codec_name.toLowerCase() === 'dvd_subtitle'
```

Comment:
Commented out dvd_subtitle?

Reply:
I haven't been able to confirm if it's absolutely necessary to remove those. I was running it with them commented out to see if anything errored during transcode. So far nothing has, but I'm also not sure if any of the 20k+ transcodes I've done have had a dvd_subtitle stream...